src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.TrackedGeneratorDriver.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class SolutionState
+    {
+        // PROTOTYPE: this mostly exists as a way to track if a GeneratorDriver has been mutated in a way
+        // which TryApplyEdit can no longer be used. We should just revise GeneratorDriver to better handle
+        // this pattern.
+        public readonly struct TrackedGeneratorDriver
```

Comment:
I think I understand the intent of this class, but it doesn't actually seem to be used? If we're not checking the NeedsFullGeneration property, why is it needed?

Reply:
Ah I just saw your comments in response to Jared. The 'posion' api is trivial. Do we want to add it into this PR and drop the tracker here instead?

Reply:
Yes, I tried to use it and then realized I have further problems to sort out. 😦 So it's left in so we *can* use it, just not quite yet.

Reply:
(The problems are mostly that I need further understand how exactly the compilation tracker reuses compilations from full states, versus proper in-progress states. It's a bit funky.)

Reply:
Ah ok, makes sense

Reply:
do you guys think you'll still need this in the future?